File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        ans = list([])
        while len(strs) > 0:
            temp = list([strs[0]])
            j = 1
            while j < len(strs):
                if sorted(strs[0]) == sorted(strs[j]):
                    temp.append(strs[j])
                    strs.pop(j)
                else:
                    j += 1
            strs.pop(0)
            ans.append(temp.copy())
            temp.clear()
        return ans

File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int ans=0;
        for(int i=0; i<s.length(); i++)
        {
            cout << i << '\n';
            int add =0;
            //nieparzyste 
            while(i-add >=0 && i+add <s.length() && s[i-add] == s[i+add])
            {
                ans++;
                add++;
            }
            //parzysta
            add=0;
            while(i-add >= 0 && i+add+1 <s.length() && s[i-add] == s[i+add+1])
            {
                ans++;
                add++;
            }
        }
        return ans;
    }
};
